java/src/main/java/com/google/oak/client/OakClient.java

```diff
@@ -16,67 +16,95 @@
+        getEvidenceResult.success().get().getAttestationEvidence();
+    AttestationEndorsement attestationEndorsement =
+        getEvidenceResult.success().get().getAttestationEndorsement();
 
-  // The following functional interfaces could have individually been replaced by a Supplier<T>, but
-  // to allow a single class implement more than one of these interfaces, dedicated functional
-  // interfaces are preferred.
+    Result<Boolean, Exception> verifyResult =
+        verifier.verify(attestationEvidence, attestationEndorsement);
+    if (getEvidenceResult.isError()) {
+      return Result.error(new Exception(getEvidenceResult.error().get()));
+    }
 
-  /** An interface for providing instances of {@code RpcClient}. */
-  public interface RpcClientProvider<C extends RpcClient> {
-    /**
-     * @return an instance of {@code RpcClient} wrapped in a {@code Result}
-     */
-    Result<C, Exception> getRpcClient();
+    OakClient<E> oakClient =
+        new OakClient<E>(transport, attestationEvidence.getEncryptionPublicKey().toByteArray());
+    return Result.success(oakClient);
   }
 
-  /** An interface for providing {@code Encryptor} instances. */
-  public interface EncryptorProvider<E extends Encryptor> {
-    /**
-     * @param signingPublicKey signing public key of the server
-     * @return an instance of Encryptor wrapped in a {@code Result}
-     */
-    Result<E, Exception> getEncryptor(byte[] signingPublicKey);
+  private OakClient(T transport, byte[] serverEncryptionPublicKey) {
+    this.transport = transport;
+    this.serverEncryptionPublicKey = serverEncryptionPublicKey;
   }
 
   /**
-   * An interface for providing instances of Evidence.
+   * Invoke the provided method by fetching and verifying the attested enclave public key, and then
+   * using it to encrypt the request body.
    *
-   * <p>An evidence normally includes the public key part of the server's signing key, an instance
-   * of {@code EndorsementEvidence}, and optionally some server configuration information. If the
-   * client policy requires verification of the server configuration, then the client should be
-   * built with a provider that does provide server configuration.
+   * @param requestBody request byte representation.
    */
-  public interface EvidenceProvider<E extends Evidence> {
-    /**
-     * Returns evidence about the trustworthiness of a remote server.
-     *
-     * @return the evidence wrapped in a {@code Result}
-     */
-    Result<E, Exception> getEvidence();
+  // TODO(#3466): Actually implement attestation and encryption.
+  public Result<byte[], Exception> invoke(byte[] requestBody) {
+    ClientEncryptor encryptor = new ClientEncryptor(this.serverEncryptionPublicKey);
+
+    // Encrypt request.
+    Result<byte[], Exception> encryptResult = encryptor.encrypt(requestBody, EMPTY_ASSOCIATED_DATA);
+    if (encryptResult.isError()) {
+      return Result.error(new Exception(encryptResult.error().get()));
```

Comment:
Uhmm this back and forth with Result and Exception does not seem very ergonomic after all. Did we get any input from some Android team on what best practices are there in terms of error handling these days? cc @rbehjati